### DISCUSSION TITLE
Development and branching strategy

### DISCUSSION THREAD

**Discussion by ioquatix** [libpng/libpng#6] [General]
I'm trying to get my head around the branching strategy that's been used. Unfortunately Glenn passed away some years ago and there is little documentation about the development process or exactly how we get from where we are to a 1.7 release. There is lots of interesting work in the 1.7 branch, but it's not clear what the testing process or goals were or what the outstanding milestones for 1.7 were.

@ctruta might be able to provide some insight here.

Current plan:

- I've renamed `master` -> `main` for consistency's sake.
- I'm merging PRs & changes into `main`. `main` is a direct descendent of `libpng16`.
- We can consider doing a (final) release of `libpng16` with all recent improvements.
- We need to merge all (reasonable) changes of `main`/`libpng16` into `libpng17` branch.

`libpng17` diverges from `libpng16` in 2012... and I'm not sure there has been any recent work for alignment.

```
$ git merge-base libpng16 libpng17
5c2d76fdef08cd01354cba9fc3ce97bca94c1499
```

This is a hard job because we need to identify what features, if any, are on `libpng16` that should be applied to the `libpng17` branch. There is literally ~10 years of commits that we need to sort out, at least that's my initial assessment.

cc @hkctkuy @thealberto @glebm @mschlaegl

**Comment by jbowler** [comment]
libpng17 was the development branch and it is a fairly complete rewrite of the internals (while maintaining API compatibility).  It got destroyed by merge.  I've tried several times to merge my pending changes into glennrp, now pnggroup, but it's a nightmare.  (Something to do with git not being able to tell the difference between a spelling correction and a code change...)

It's moot now: Cosmin made the future semi-clear here:

https://sourceforge.net/p/png-mng/mailman/message/58818920/

He hasn't commented on any of the comments so I have to assume it's a done deal.  libpng-1.7 is pretty much "libpng-ng"; i.e. now dead because of the next episode (as in the base Star Trek reference.)

Perhaps you would care to comment on png-mng-implement?
